tests/test_modeling_common.py

```diff
@@ -3132,13 +3130,355 @@ def test_flash_attn_2_generate_padding_right(self):
 
                 self.assertTrue(torch.allclose(out, out_fa))
 
+    @parameterized.expand([("float16",), ("bfloat16",), ("float32",)])
+    @require_torch_sdpa
+    @slow
+    def test_eager_matches_sdpa_inference(self, torch_dtype: str):
```

Comment:
Ok for now, please refactor this in a followup PR to just make it a bit readable with some function splits / split the tests with sub testing whatever but it's too long and will be hard to debug 😉